### PR TITLE
Renamed INode.RangeToken property to .Origin

### PR DESCRIPTION
**Pull request by keyboardDrummer**
This PR extracts changes from https://github.com/dafny-lang/dafny/pull/5931 to make that one simpler to review

### What was changed?
- Renamed `INode.RangeToken` property to .Origin. 
- Extracted a few classes from Tokens.cs to separate files
- Renamed a few methods in Cloner.cs

### How has this been tested?
Pure refactoring

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: .github/workflows/doc-tests.yml [hunk 1]
```diff
@@ -30,6 +30,10 @@ jobs:
       uses: actions/setup-dotnet@v4
       with:
         dotnet-version: 6.0.x
```
Comment (robin-aws) [inline .github/workflows/doc-tests.yml:32]: Probably want to delete the first copy :)

File: .github/workflows/runtime-tests.yml [hunk 2]
```diff
@@ -25,6 +28,14 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: true
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{env.dotnet-version}}
```
Comment (robin-aws) [inline .github/workflows/runtime-tests.yml:34]: Definitely delete the first copy (and the useless `env`)!

**Review by robin-aws — CHANGES_REQUESTED** [review]
LGTM and thank you for the build fix, sorry to be a stickler on the GHA workflow typos but they are very confused as is.

File: .github/workflows/runtime-tests.yml (hunk 2 above)
Comment (keyboardDrummer) [reply .github/workflows/runtime-tests.yml:34]: What do you mean? I'm purposefully installing both 8 and 6, so I'm invoking `actions/setup-dotnet@v4` twice

File: .github/workflows/doc-tests.yml (hunk 1 above)
Comment (keyboardDrummer) [reply .github/workflows/doc-tests.yml:32]: I've tried various combinations and the one that works is to install both 6 and 8

Comment (robin-aws) [reply .github/workflows/doc-tests.yml:32]: Oh! So it's something about building and running with different versions?

In that case add comments instead. As is it just looks unintentional and someone will 100% remove it later.

**Review by robin-aws — APPROVED** [review]

**Pull request merged**